packages/@o3r/rules-engine/builders/rules-engine-extractor/helpers/rules-engine.extractor.interfaces.ts

```diff
@@ -1,18 +1,49 @@
 /** Supported types for a fact */
-export type FactSupportedTypes = 'string' | 'number' | 'date' | 'object' | 'boolean';
+export type FactSupportedTypes = 'string' | 'number' | 'date' | 'object' | 'boolean' | 'array';
 
-/** Fact as description in the metadata file */
-export interface MetadataFact {
+/** Base metadata fact */
+interface BaseMetadataFact {
```

Comment:
There is a reason to not export this interface?

Reply:
It is not being used outside of this file so I removed the export

Reply:
Are you sure it is not used by our users?

Reply:
I don't think so since I just created this interface but I can export it in case it needs to be used

Reply:
My bad, I was just looking at the comment and did not realize you were not talking about the MetadataFact which is still exposed as an union type. It is fine for me.